IPython/terminal/interactiveshell.py

```diff
@@ -439,8 +439,14 @@ def mainloop(self, display_banner=DISPLAY_BANNER_DEPRECATED):
             try:
                 self.interact()
                 break
-            except KeyboardInterrupt:
-                print("\nKeyboardInterrupt escaped interact()\n")
+            except KeyboardInterrupt as e:
+                print("\n%s escaped interact()\n" % type(e).__name__)
+                # An interrupt during the eventloop will mess up the
+                # internal state of the prompt_toolkit library.
+                # Stopping the eventloop fixes this, see
+                # https://github.com/ipython/ipython/pull/9867
```

Comment:
Just to check, it's safe to call `.stop()` on an already-stopped eventloop, right?